### PR TITLE
Use cudf::thread_index_type in get_json_object and tdigest kernels

### PR DESCRIPTION
**Pull request by nvdbaranec**
Convert the grid-stride loop in `get_json_object_kernel` to use `cudf::thread_index_type`.  Convert `compute_percentiles_kernel` to use `cudf::thread_index_type`.  

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by bdice — APPROVED** [review]

File: cpp/src/quantiles/tdigest/tdigest.cu [hunk 1]
```diff
@@ -74,7 +74,8 @@ __global__ void compute_percentiles_kernel(device_span<size_type const> tdigest_
                                            double const* cumulative_weight_,
                                            double* output)
 {
-  int const tid = threadIdx.x + blockIdx.x * blockDim.x;
+  auto const tid = cudf::thread_index_type{threadIdx.x} +
+                   (cudf::thread_index_type{blockIdx.x} * cudf::thread_index_type{blockDim.x});
```
[the remark this answers is not in the document]
Comment (mythrocks) [reply cpp/src/quantiles/tdigest/tdigest.cu:78]: TIL

**Review by mythrocks — APPROVED** [review]
LGTM. It would be good to incorporate the `global_thread_id()` change.

File: cpp/src/quantiles/tdigest/tdigest.cu (hunk 1 above)
[the remark this answers is not in the document]
Comment (hyperbolic2346) [reply cpp/src/quantiles/tdigest/tdigest.cu:78]: I did some checking so I didn't feel so in the dark. David Wendt said this was added last week, so I don't feel so badly about not knowing about it. Very useful.

**Comment by davidwendt** [comment]
Looks like someone did not set up their github account to support signed commits.

**Review by karthikeyann — APPROVED** [review]

**Comment by ajschmidt8** [comment]
admin merging due to the `/merge` comment and since the last commit was empty and the commit before that passed CI.

**Comment by ajschmidt8** [comment]
admin merging due to the `/merge` command and since the last commit was empty, but the commit before that passed CI.

**Pull request merged**